delegator.py

```diff
@@ -55,15 +56,18 @@ def std_out(self):
 
     @property
     def _pexpect_out(self):
-        result = ''
+        if self.subprocess.encoding:
+            result = ''
+        else:
+            result = bytes()
 
         if self.subprocess.before:
             result += self.subprocess.before
 
         if isinstance(self.subprocess.after, str):
```

Comment:
So it seems like this will have issues between Python 2 and 3, where Python 2 recognizes a bytes object as a `str` but Python 3 won't. I'm assuming if we pass an `encoding` of `None` to PopenSpawn that everything returned is bytes but could you confirm that for `before` and `after`? If it is, let's change the conditional to match the above (`if self.subprocess.after`).